apps/api/src/app/environments/environments.bridge.controller.ts

```diff
@@ -0,0 +1,161 @@
+import {
+  Body,
+  ClassSerializerInterceptor,
+  Controller,
+  Get,
+  NotFoundException,
+  Param,
+  Post,
+  Query,
+  Req,
+  Res,
+  UseInterceptors,
+} from '@nestjs/common';
+import { Request, Response } from 'express';
+import { ApiTags } from '@nestjs/swagger';
+import { decryptApiKey } from '@novu/application-generic';
+import {
+  EnvironmentRepository,
+  NotificationStepEntity,
+  NotificationTemplateEntity,
+  NotificationTemplateRepository,
+} from '@novu/dal';
+import { Client, Event, Step, Workflow, workflow, WorkflowChannelEnum } from '@novu/framework';
+import { PreviewResult, StepTypeEnum, TRANSIENT_PREVIEW_PREFIX } from '@novu/shared';
+import { ApiCommonResponses } from '../shared/framework/response.decorator';
+import { NovuNestjsHandler } from './novu-nestjs-handler';
+import { OutputRendererFactory } from './render/output-render-factory';
+
+@ApiCommonResponses()
+@Controller('/environments')
+@UseInterceptors(ClassSerializerInterceptor)
+@ApiTags('Environments')
+export class EnvironmentsBridgeController {
+  constructor(
+    private workflowsRepository: NotificationTemplateRepository,
+    private environmentsRepository: EnvironmentRepository
+  ) {}
+
+  @Get('/:environmentId/bridge')
+  async getHandler(@Req() req: Request, @Res() res: Response, @Param('environmentId') environmentId: string) {
+    /*
+     * TODO: remove the GET handler, it's only used right now for the bridge health check,
+     * and it's not needed for UI-defined workflows.
+     *
+     * I'm leaving it here as a reminder that we need to remove the UI health-check for Novu-managed
+     * bridge environments.
+     */
+    const novuBridgeHandler = new NovuNestjsHandler({
+      workflows: [],
+      client: new Client({
+        strictAuthentication: false,
+        secretKey: await this.getApiKey(environmentId),
+      }),
+    });
+    await novuBridgeHandler.handleRequest(req, res, 'GET');
+  }
+
+  @Post('/:environmentId/bridge')
```

Comment:
i wonder if we want to state the action as well as right now it feels too generic and not self-explanatory.

```suggestion
  @Post('/:environmentId/bridge-preview')
```